xdsl/parser/base_parser.py

```diff
@@ -252,14 +252,15 @@ def parse_optional_undelimited_comma_separated_list(
         parse: Callable[[], _AnyInvT],
     ) -> list[_AnyInvT] | None:
         """
-        Parses greedily a list of elements separated by commas, and delimited
-        by the specified delimiter. Return None if no elements were parsed.
+        Parses greedily a list of elements separated by commas, if a first element is
+        present. The first element is parsed with `parse_optional`, and the remaining
+        are parsed with `parse`. The parsing stops either if the first element is not
+        present, or if no comma is present after parsing an element.
         """
```

Comment:
```python
if (elem := parse_optional()) is not None:
    elems = [elem]
    while self._parse_optional_token(Token.Kind.COMMA) is not None:
        elems.append(parse())
    return elems
return None
```

Reply:
I prefer the version in the code, the successful path is down towards the end of the function, and it avoids nesting.